aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -103,38 +103,49 @@ void _validate_sparse_compressed_tensor_args_worker(const Tensor& compressed_ind
       "number of dimensions of ", compressed_indices_name, " and ", plain_indices_name, " must be the same but got ",
       compressed_indices.dim(), " and ", plain_indices.dim(), ", respectively");
 
-  AT_DISPATCH_PLAIN_SPARSE_COMPRESSED_LAYOUTS(
+  int block_ndim = AT_DISPATCH_PLAIN_SPARSE_COMPRESSED_LAYOUTS(
```

Comment:
For follow up: I'm wondering whether we should store this as a member variable of SparseCsrTensorImpl.h. Really its value can only be 0 or 2.

Reply:
Currently, the `blocksize` can be uniquely determined from the sparse tensor `layout` and the shape of `values` tensor (`blocksize` is `values().shape[1:3]` for BSR/BSC tensors). That said, we could implement a `blocksize()` method for `SparseCsrTensorImpl` that DimVector instance of a blocksize according to the layout and values info.

Reply:
I suggest that we address this when implementing hybrid sparse compressed tensors support.

Reply:
Although, when calling `validate_sparse_compressed_tensor_args` function, our inputs are the arguments to sparse compressed tensor factory functions. So, introducing `blocksize()` method for SparseCsrTensorImpl is out of scope of the function.